corehq/apps/case_search/fixtures.py

```diff
@@ -0,0 +1,75 @@
+from lxml.builder import E
+
+from casexml.apps.phone.fixtures import FixtureProvider
+
+from corehq import extensions
+from corehq.apps.case_search.exceptions import CaseFilterError
+from corehq.apps.case_search.filter_dsl import build_filter_from_xpath
+from corehq.apps.es.case_search import CaseSearchES
+from corehq.messaging.templating import (
+    MessagingTemplateRenderer,
+    NestedDictTemplateParam,
+)
+from corehq.toggles import MODULE_BADGES
+
+from .models import CSQLFixtureExpression
+
+
+def _get_user_template_info(restore_user):
+    return {
+        "username": restore_user.username,
+        "uuid": restore_user.user_id,
+        "user_data": restore_user.user_session_data,
+        "location_ids": restore_user.get_location_ids(restore_user.domain),
+    }
+
+
+def _get_template_renderer(restore_user):
+    renderer = MessagingTemplateRenderer()
+    renderer.set_context_param('user', NestedDictTemplateParam(_get_user_template_info(restore_user)))
+    for name, param in custom_csql_fixture_context(restore_user.domain, restore_user):
+        renderer.set_context_param(name, param)
+    return renderer
+
+
+@extensions.extension_point
+def custom_csql_fixture_context(domain, restore_user):
+    '''Register custom template params to be available in CSQL templates'''
```

Comment:
A reference to `corehq/apps/app_manager/suite_xml/post_process/instances.py` would be nice for future devs reading this code.

Reply:
Can you elaborate?  They shouldn't need to register anything else there if extending this work.  I did consider importing `CaseSearchFixtureProvider` over there to reference the `id`, but I see we don't do that for the other fixtures, and I didn't want to introduce weird import dependencies.

Reply:
Copied the link for the wrong file. I meant the settings. But thinking more about it, it is pretty easy to follow the names to get there.